src/helpers/merge_page_data.ts

```diff
@@ -0,0 +1,10 @@
+import { Actor } from "apify";
+
+// helps merge new page data into previous pages already stored in KVStore
+export const mergePageData = async (key: string, newData: unknown[]) => {
+    const prevData = await Actor.getValue<unknown[]>(key);
+    if (prevData) {
+        newData = [...prevData, newData];
+    }
+    await Actor.setValue(key, newData);
+};
```

Comment:
This is usually an anti-pattern but it depends a lot on the scale. The problem with this function is that each `getValue` and `setValue` is an API call. In case there are thousands of very fast requests, this could reach the API limit.

Generally, we the "persistedState" pattern which works with in-memory value only stored to KV store for persistence. That has its own drawbacks that it can get too big. There are 2 ways to handle that:
1. Once certain data gets complete, the whole key should be deleted from the state
2. If it gets huge even with that, it can sharded in the KV store. It can be stored & loaded in multiple records and then merged into single object so it loads quickly.

Again, we don't need to rewrite this now so perhaps just add this as a comment once we need to resolve this scale.